### DISCUSSION TITLE
Statefull vs Stateless Performance

### DISCUSSION THREAD

**Discussion by conquera99** [vercel/next.js#17421] [Help] [answered]
is there any performance difference between stateful and stateless component?

is there any speciality that cannot be used with stateful but can be used with stateless and vice versa?

thank you

**Comment by lfades** [comment, marked as answer]
> is there any performance difference between stateful and stateless component?

**No.** Using a stateful or stateless component won't affect your performance so you don't to worry about it. What you're rendering is what affects your performance.

> is there any specialty that cannot be used with stateful but can be used with stateless and vice versa?

**No.**

Something I recommend is using Functional Components over Classes, that can make a difference as it makes you more bulletproof for future React improvements and releases

**Reply by conquera99** [reply]
okay thank you @lfades thats why people always use functional components.
